In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import scipy.stats as stats

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list the files in the input directory

import os
print(os.listdir("../input"))

# Any results you write to the current directory are saved as output.

['Cust_demo.csv', 'sample_survey.csv', 'Credit Card Data.xlsx', 'spend.csv', 'creditpromo.csv', 'Customer.csv', 'dvdplayer.csv', 'SalesData.csv', 'Testing_Of_Hypothesis.csv', 'Cust_data.csv', 'cust_demographics.csv', 'appl_1980_2014.csv', 'Transactions.csv', 'cust_new.csv', 'Repayment.csv', 'Customers.csv', 'Customer Acqusition.csv', 'prod_cat_info.csv', 'pollination.csv', 'claims.csv', 'tolerance.csv', 'dietstudy.csv']


In [2]:
diet = pd.read_csv('../input/dietstudy.csv')
diet.head()

patid  age  gender  tg0  tg1  tg2  ...   tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  ...   100   198   196   193   188   192
1      2   56    Male  139   94  119  ...    92   237   233   232   228   225
2      3   50    Male  152  185   86  ...   118   233   231   229   228   226
3      4   46  Female  112  145  136  ...    82   179   181   177   174   172
4      5   64    Male  156  104  157  ...    97   219   217   215   213   214

[5 rows x 13 columns]

# CI : 0.95 or 95%
# alpha = 0.05
# if p value < alpha then reject NULL and 
# if it is >= alpha, we fail to reject NULL

In [3]:
print('pre diet weight: ', diet.wgt0.mean())
print('pre diet triglyceride level: ', diet.tg0.mean())

pre diet weight:  198.375
pre diet triglyceride level:  138.4375


In [4]:
print('post 1 month diet weight: ', diet.wgt1.mean())
print('post 1 month diet triglyceride level: ', diet.tg1.mean())

post 1 month diet weight:  196.125
post 1 month diet triglyceride level:  124.5625


### First month
### Ho: weight before diet == weight after 1 month (no change)
### Ha: weight before diet != weight after 1 month (change)

### Ho: triglyceride before diet == triglyceride after 1 month 
### Ha: triglyceride before diet != triglyceride after 1 month

In [5]:
month1_wt = stats.ttest_rel(a=diet.wgt0, b=diet.wgt1)
month1_tg = stats.ttest_rel(a=diet.tg0, b=diet.tg1)

In [6]:
print(month1_wt.pvalue < 0.05)
print(month1_tg.pvalue < 0.05)

True
False


### Conclusion: Afer 1 month of dieting, there is a change in weight but there is no significant change in triglyceride level.

### Second month
### Ho: weight before diet == weight after 2 month
### Ha: weight before diet != weight after 2 month

### Ho: triglyceride before diet == triglyceride after 2 month
### Ha: triglyceride before diet != triglyceride after 2 month

In [7]:
print('post 2 month diet weight: ', diet.wgt2.mean())
print('post 2 month diet triglyceride level: ', diet.tg2.mean())

post 2 month diet weight:  194.125
post 2 month diet triglyceride level:  124.375


In [8]:
month2_wt = stats.ttest_rel(a=diet.wgt0, b=diet.wgt2)
month2_tg = stats.ttest_rel(a=diet.tg0, b=diet.tg2)

In [9]:
print(month2_wt.pvalue < 0.05)
print(month2_tg.pvalue < 0.05)

True
False


### Conclusion: Afer 2 months of dieting, there is a change in weight but there is no significant change in triglyceride level.

# -----------------------------------------------------------------------

### Third month
### Ho: weight before diet == weight after 3 months
### Ha: weight before diet != weight after 3 months

### Ho: triglyceride before diet == triglyceride after 3 months
### Ha: triglyceride before diet != triglyceride after 3 months

In [10]:
print('post 3 months diet weight: ', diet.wgt3.mean())
print('post 3 months diet triglyceride level: ', diet.tg3.mean())

post 3 months diet weight:  192.125
post 3 months diet triglyceride level:  118.8125


In [11]:
month3_wt = stats.ttest_rel(a=diet.wgt0, b=diet.wgt3)
month3_tg = stats.ttest_rel(a=diet.tg0, b=diet.tg3)

In [12]:
print(month3_wt.pvalue < 0.05)
print(month3_tg.pvalue < 0.05)

True
False


### Conclusion: Afer 3 months of dieting, there is a change in weight but there is no significant change in triglyceride level.

# ---------------------------------------------------------------------

### Fourth month
### Ho: weight before diet == weight after 4 months
### Ha: weight before diet != weight after 4 months

### Ho: triglyceride before diet == triglyceride after 4 months
### Ha: triglyceride before diet != triglyceride after 4 months

In [13]:
print('post 4 months diet weight: ', diet.wgt4.mean())
print('post 4 months diet triglyceride level: ', diet.tg4.mean())

post 4 months diet weight:  190.3125
post 4 months diet triglyceride level:  124.375


In [14]:
month4_wt = stats.ttest_rel(a=diet.wgt0, b=diet.wgt4)
month4_tg = stats.ttest_rel(a=diet.tg0, b=diet.tg4)

In [15]:
print(month4_wt.pvalue < 0.05)
print(month4_tg.pvalue < 0.05)

True
False


### Conclusion: Afer 4 months of dieting, there is a change in weight but there is no significant change in triglyceride level.

# -----------------------------------------------------------------------

> **Credit Promo**

1. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales?

In [16]:
credit = pd.read_csv('../input/creditpromo.csv')
credit.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

In [17]:
standard = credit.dollars[credit['insert']=='Standard']
new_promo = credit.dollars[credit['insert']=='New Promotion']

In [18]:
print('Spend of customer with standard promo: ', standard.mean())
print('Spend of customer with new promo: ', new_promo.mean())

Spend of customer with standard promo:  1566.3890309659348
Spend of customer with new promo:  1637.4999830647992


### Ho: standard promo spend == new promo spend (promo failure)
### Ha: standard promo spend != new promo spend (promo success)


In [19]:

credit_equalv = stats.ttest_ind(a= standard, b=new_promo, equal_var=True)
credit_unequalv = stats.ttest_ind(a= standard, b=new_promo, equal_var=False)

In [20]:
credit_equalv.statistic - credit_unequalv.statistic

-4.440892098500626e-16

In [21]:
credit_equalv.pvalue < 0.05

True

### Conclusion: Customers who received new promotion spend more than the customers with standard promo.
### Hence, promotion is effective.

# -----------------------------------------------------------------------

# > **Pollination**
1. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv) a. Is the overall population of Seed yield/plant (g) equals to 200? b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.

In [22]:
pol = pd.read_csv('../input/pollination.csv')
pol.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

In [23]:
pol.Seed_Yield_Plant.mean()


180.8035

### Ho: Seed yield/plant == 200
### Ha: Seed yield/plant <  200

In [24]:
seed_yield = stats.ttest_1samp(a=pol.Seed_Yield_Plant, popmean=200)

In [25]:
seed_yield.pvalue < 0.05

True

### Conclusion: Seed yield/plant is not equal to 200, it is lower.

In [26]:
natural_pol = pol.loc[pol.Group == 'Natural']

hand_pol = pol.loc[pol.Group == 'Hand']


In [27]:
natural_pol.Seedling_length.mean()

17.706999999999994

In [28]:
hand_pol.Seedling_length.mean()

18.589999999999996

#Ho: Seedling_length from natural pollination == Seedling_length from hand pollination (equally effective)
#Ha: Seedling_length from natural pollination != Seedling_length from hand pollination (significantly different)

In [29]:
seed_length = stats.ttest_ind(a=natural_pol.Seedling_length, b=hand_pol.Seedling_length)
seed_length

Ttest_indResult(statistic=-2.542229999657055, pvalue=0.020428817064110226)

In [30]:
seed_length.pvalue

0.020428817064110226

In [31]:
print(natural_pol.Fruit_Wt.mean())
print(hand_pol.Fruit_Wt.mean())

print(natural_pol.Seed_Yield_Plant.mean())
print(hand_pol.Seed_Yield_Plant.mean())

1.848
2.5660000000000003
146.009
215.598


In [32]:
fruit_wt = stats.ttest_ind(a=natural_pol.Fruit_Wt, b=hand_pol.Fruit_Wt)
seed_yield = stats.ttest_ind(a=natural_pol.Seed_Yield_Plant, b=hand_pol.Seed_Yield_Plant)

In [33]:
fruit_wt.pvalue

8.078362076486221e-13

In [34]:
seed_yield.pvalue

4.2714815854843853e-11

### Conclusion: As p-value is less than 0.05, we reject null hypothesis, there is a significant differenc in natural pollination and hand pollination under open field conditions.

# DVD player
1. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently?

In [35]:
dvd = pd.read_csv('../input/dvdplayer.csv')
dvd.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

In [36]:
dvd.agegroup.value_counts()

65 and over    17
Under 25       13
35-44          12
45-54          10
25-34          10
55-64           6
Name: agegroup, dtype: int64

In [37]:
grp1 = dvd.dvdscore.loc[dvd.agegroup == 'Under 25']
grp2 = dvd.dvdscore.loc[dvd.agegroup == '25-34']
grp3 = dvd.dvdscore.loc[dvd.agegroup == '35-44']
grp4 = dvd.dvdscore.loc[dvd.agegroup == '45-54']
grp5 = dvd.dvdscore.loc[dvd.agegroup == '55-64']
grp6 = dvd.dvdscore.loc[dvd.agegroup == '65 and over']

### Ho: mean grp1 == mean grp2 == mean grp3 == mean grp4 == mean grp5 == mean grp6 (rated equally similar)
### Ha: mean grp1 != mean grp2 != mean grp3 != mean grp4 != mean grp5 != mean grp6 (rated differently)

In [38]:
dvd_anova = stats.f_oneway(grp1, grp2, grp3, grp4, grp5, grp6)
dvd_anova

F_onewayResult(statistic=6.992526962676518, pvalue=3.087324905679639e-05)

### Conclusion: Consumers of different age groups rated the design differently.

## Sample Survey
A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv). a. Is there any relationship in between labour force status with marital status? b. Do you think educational qualification is somehow controlling the marital status? c. Is happiness is driven by earnings or marital status?

In [39]:
sample_data = pd.read_csv('../input/sample_survey.csv')
sample_data.head(2)

id            wrkstat        marital    ...         car1      car2      car3
0   1  Working full time       Divorced    ...     American  Japanese  Japanese
1   2  Working part-time  Never married    ...     American    German  Japanese

[2 rows x 46 columns]

In [40]:
sample_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2832 entries, 0 to 2831
Data columns (total 46 columns):
id          2832 non-null int64
wrkstat     2831 non-null object
marital     2831 non-null object
childs      2825 non-null float64
age         2828 non-null float64
educ        2820 non-null float64
paeduc      2041 non-null float64
maeduc      2399 non-null float64
speduc      1311 non-null float64
degree      2822 non-null object
sex         2832 non-null object
race        2832 non-null object
born        2819 non-null object
parborn     2812 non-null object
granborn    2630 non-null float64
income      2503 non-null object
rincome     1849 non-null object
polviews    2691 non-null object
cappun      2599 non-null object
postlife    2066 non-null object
happy       2806 non-null object
hapmar      1337 non-null object
owngun      1869 non-null object
news        1870 non-null object
tvhours     2337 non-null float64
howpaid     1347 non-null object
ethnic      2238 non-null ob

### Ho: Observed == Expected (no influence)
### Ha: Observed != Expected (influence)

In [41]:
wrk_mar_xtab = pd.crosstab(sample_data.wrkstat, sample_data.marital, margins=True)
wrk_mar_xtab

marital                  Divorced  Married  ...   Widowed   All
wrkstat                                     ...                
Keeping house                  25      200  ...        55   328
Other                          12       16  ...         8    54
Retired                        53      168  ...       150   394
School                          7        9  ...         1    79
Temporarily not working         9       23  ...         2    46
Unemployed, laid off           10       13  ...         3    58
Working full time             295      778  ...        44  1567
Working part-time              35      138  ...        20   304
All                           446     1345  ...       283  2830

[9 rows x 6 columns]

In [42]:
wrk_mar_test = stats.chi2_contingency(observed=wrk_mar_xtab)
wrk_mar_test

(729.2421426572284,
 1.820339965538765e-127,
 40,
 array([[5.16918728e+01, 1.55886926e+02, 7.68424028e+01, 1.07787986e+01,
         3.28000000e+01, 3.28000000e+02],
        [8.51024735e+00, 2.56643110e+01, 1.26508834e+01, 1.77455830e+00,
         5.40000000e+00, 5.40000000e+01],
        [6.20932862e+01, 1.87254417e+02, 9.23045936e+01, 1.29477032e+01,
         3.94000000e+01, 3.94000000e+02],
        [1.24501767e+01, 3.75459364e+01, 1.85077739e+01, 2.59611307e+00,
         7.90000000e+00, 7.90000000e+01],
        [7.24946996e+00, 2.18621908e+01, 1.07766784e+01, 1.51166078e+00,
         4.60000000e+00, 4.60000000e+01],
        [9.14063604e+00, 2.75653710e+01, 1.35879859e+01, 1.90600707e+00,
         5.80000000e+00, 5.80000000e+01],
        [2.46954770e+02, 7.44740283e+02, 3.67109894e+02, 5.14950530e+01,
         1.56700000e+02, 1.56700000e+03],
        [4.79095406e+01, 1.44480565e+02, 7.12197880e+01, 9.99010601e+00,
         3.04000000e+01, 3.04000000e+02],
        [4.46000000e+02, 1.345

### Conclusion:  labour force status have relationship with marital status

In [43]:
degree_mar_xtab = pd.crosstab(sample_data.degree, sample_data.marital, margins=True)
degree_mar_xtab

marital         Divorced  Married  Never married  Separated  Widowed   All
degree                                                                    
Bachelor              58      251            129         12       28   478
Graduate              29      123             41          3        9   205
High school          241      686            367         58      148  1500
Junior college        45      108             46          3        6   208
LT High school        70      174             77         17       92   430
All                  443     1342            660         93      283  2821

In [44]:
degree_mar_test = stats.chi2_contingency(observed=degree_mar_xtab)
degree_mar_test[1]

7.424404099753273e-15

### Conclusion: Educational qualification has influence on the marital status.

In [45]:
happy_mar_xtab = pd.crosstab(sample_data.happy, sample_data.marital, margins=True)
happy_mar_xtab

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        278      684            426         49      137  1574
Very happy           93      582            120         13       83   891
All                 443     1337            654         92      279  2805

In [46]:
happy_mar_test = stats.chi2_contingency(observed=happy_mar_xtab)
happy_mar_test

(260.68943894182826,
 7.762777322980048e-47,
 15,
 array([[  53.6969697 ,  162.06060606,   79.27272727,   11.15151515,
           33.81818182,  340.        ],
        [ 248.58538324,  750.24527629,  366.98609626,   51.62495544,
          156.55828877, 1574.        ],
        [ 140.71764706,  424.69411765,  207.74117647,   29.22352941,
           88.62352941,  891.        ],
        [ 443.        , 1337.        ,  654.        ,   92.        ,
          279.        , 2805.        ]]))

In [47]:
happy_income_xtab = pd.crosstab(sample_data.happy, sample_data.income, margins=True)
happy_income_xtab

income         $1000 TO 2999  $10000 - 14999  ...   LT $1000   All
happy                                         ...                 
Not too happy              7              39  ...         11   302
Pretty happy              20             107  ...         13  1408
Very happy                 5              44  ...         11   771
All                       32             190  ...         35  2481

[4 rows x 13 columns]

In [48]:
happy_income_test = stats.chi2_contingency(observed=happy_income_xtab)
happy_income_test

(178.95053061216427,
 7.234749067043371e-21,
 36,
 array([[   3.89520355,   23.12777106,   21.66706973,   29.82265216,
          191.35187424,    2.92140266,    3.89520355,    4.26037888,
            4.01692866,    5.72108021,    7.06005643,    4.26037888,
          302.        ],
        [  18.16041919,  107.82748892,  101.01733172,  139.04070939,
          892.1305925 ,   13.62031439,   18.16041919,   19.86295848,
           18.72793229,   26.67311568,   32.91575977,   19.86295848,
         1408.        ],
        [   9.94437727,   59.04474002,   55.31559855,   76.13663845,
          488.51753325,    7.45828295,    9.94437727,   10.87666264,
           10.25513906,   14.60580411,   18.0241838 ,   10.87666264,
          771.        ],
        [  32.        ,  190.        ,  178.        ,  245.        ,
         1572.        ,   24.        ,   32.        ,   35.        ,
           33.        ,   47.        ,   58.        ,   35.        ,
         2481.        ]]))

### Conclusion: As p-value for both cases (happiness with marital status vs. happiness with earnings) is less, 
### both has influence on happiness, but as chi-square test score is more with marital status (260) as compared with earnings (178),
### happiness is driven more by marital status.